build.sbt

```diff
@@ -103,19 +106,25 @@ lazy val commonSettings = Seq(
   // format: on
 )
 
-// add support for Scala version ranges such as "scala-2.12+" in source folders (single version folders such as
-// "scala-2.11" are natively supported by SBT).
-// In order to keep this simple, we're doing this case by case, taking advantage of the fact that we intend to support
-// only 3 major versions at any given moment.
+// add support for Scala version ranges such as "scala-2.12+" or "scala-2.13-" in source folders (single version folders
+// such as "scala-2.11" are natively supported by SBT).
 def crossVersionSharedSources(unmanagedSrcs: SettingKey[Seq[File]]) = {
   unmanagedSrcs ++= {
-    val minor = CrossVersion.partialVersion(scalaVersion.value).map(_._2)
-    List(
-      if (minor.exists(_ <= 12)) unmanagedSrcs.value.map { dir => new File(dir.getPath + "-2.12-") }
-      else Nil,
-      if (minor.exists(_ >= 12)) unmanagedSrcs.value.map { dir => new File(dir.getPath + "-2.12+") }
-      else Nil
-    ).flatten
+    val version = CrossVersion.partialVersion(scalaVersion.value)
+    val expectedVersions = Seq((2, 11), (2, 12), (2, 13), (3, 0))
+    expectedVersions.flatMap { case (major, minor) =>
+      lazy val versionV = major * 100 + minor
```

Comment:
Scala provides an `Ordering[(Int, Int)]` instance. Can we compare pairs directly for conciseness and to avoid introducing assumptions on how versioning is done?

Reply:
Better yet, SBT provides native support for semantic versioning: https://www.scala-sbt.org/1.0/docs/sbt-1.2-Release-Notes.html#Semantic+Version+selector+API

Reply:
Nice! Unfortunately, the semantic version support doesn't seem to play very well with versions with a suffix (like the milestone release of Scala 3 "3.0.0-M2"). For the time being, I ended up using the `Ordering` instance for this.